File: Selenium/Project/Activity6.py
from selenium import webdriver
from selenium.webdriver.common.by import By

# Create an instance of the Firefox webdriver
driver = webdriver.Firefox()

# Open the webpage
driver.get("https://alchemy.hguy.co/lms")

# Find the "My Account" link element using XPath and click on it
driver.find_element(By.XPATH, "//a[contains(text(),'My Account')]").click()

# Check if the title of the page is "My Account – Alchemy LMS"
assert driver.title == "My Account – Alchemy LMS"

# Find the "Login" link element using XPath and click on it
driver.find_element(By.XPATH, "//a[contains(text(),'Login')]").click()

# Find the username input field using XPath and enter the username
driver.find_element(By.XPATH, "//p[@class='login-username']/input").send_keys("root")

# Find the password input field by ID and enter the password
driver.find_element(By.ID, "user_pass").send_keys("pa$$w0rd")

# Find the login button by ID and click on it
driver.find_element(By.ID, ("wp-submit")).click()

# Check if the "Logout" link is displayed
assert driver.find_element(By.XPATH, ("//a[contains(text(),'Logout')]")).is_displayed()

# Find the "Logout" link element using XPath and click on it
driver.find_element(By.XPATH, ("//a[contains(text(),'Logout')]")).click()

# Close the browser
driver.quit()

File: Selenium/Project/Activity6.java
import org.openqa.selenium.By;
import org.testng.Assert;
import org.testng.annotations.Test;

public class Activity6 extends BaseClass{
    @Test
    public void activity6() {
        driver.findElement(By.xpath("//a[contains(text(),'My Account')]")).click();
        Assert.assertEquals(driver.getTitle(), "My Account – Alchemy LMS");
        driver.findElement(By.xpath("//a[contains(text(),'Login')]")).click();
        driver.findElement(By.xpath("//p[@class='login-username']/input")).sendKeys("root");
        driver.findElement(By.id("user_pass")).sendKeys("pa$$w0rd");
        driver.findElement(By.id("wp-submit")).click();
        Assert.assertTrue(true, String.valueOf(driver.findElement(By.xpath("//a[contains(text(),'Logout')]")).isDisplayed()));
        driver.findElement(By.xpath("//a[contains(text(),'Logout')]")).click();
    }
}
